### PR TITLE
M90 changes

### PR DESCRIPTION
**Pull request by TJatPBnJ**
## About The Pull Request
The m90 is now 14tc instead of 18. Its grenades can be bought from the uplink in boxes for 6tc, instead of individual grenades for 2tc each. You can also fire the m90's grenade launcher with right click. 

## Why It's Good For The Game
The m90 isn't used very often, and the grenade launcher on it is currently hard to use, so having right click fire it makes it less difficult and more intuitive to use. For 18tc, it's overshadowed by the l6, which is why I lowered the cost of the gun and it's grenades.

## Changelog
:cl:
tweak: The M90 now uses right click to fire it's grenade launcher.
balance: 40mm grenades are now bought in ammo boxes for 6tc rather than individually for 2tc
/:cl:

### CODE REVIEW

**Comment by TJatPBnJ** [comment]
any feedback or suggestions would be appreciated

**Comment by Pugie** [comment]
God bless

**Comment by necromanceranne** [comment]
Right click grenade launchers, absolute genius.

**Comment by Fhaxaris** [comment]
image
it doesn't look like they have an icon

**Comment by TJatPBnJ** [comment]
hmm i was hoping that didnt do anything
ill just lower it back to 4, 8 does seem a bit much after a bit of thinking

**Comment by Son-of-Space** [comment]
A weapon never seen will now be used. Good job.

File: code/modules/projectiles/guns/ballistic/automatic.dm [hunk 1]
```diff
@@ -165,7 +165,8 @@
 	update_icon()
 
 /obj/item/gun/ballistic/automatic/m90/afterattack(atom/target, mob/living/user, flag, params)
-	if(select == 2)
+	var/list/modifiers = params2list(params)
+	if(modifiers["right"] && !Adjacent(target))
```
Comment (Jared-Fogle) [inline code/modules/projectiles/guns/ballistic/automatic.dm:169]: Use the new alt attack procs, and don't bother checking adjacency.

Comment (TJatPBnJ) [reply code/modules/projectiles/guns/ballistic/automatic.dm:169]: I was checking adjacency so you could still hold people up with the gun without blowing yourself up, are you sure?

Comment (TJatPBnJ) [reply code/modules/projectiles/guns/ballistic/automatic.dm:169]: After staring at a screen confused for several hours, I've finally found that the alt attack procs aren't even called when a gun is fired. I feel slightly less sane, thanks.

Comment (TJatPBnJ) [reply code/modules/projectiles/guns/ballistic/automatic.dm]: > I was checking adjacency so you could still hold people up with the gun without blowing yourself up, are you sure?

The staring has also made me realize why I didn't need to check adjacency.

**Review by Mothblocks — CHANGES_REQUESTED** [review]
use _secondary now

File: code/modules/projectiles/guns/ballistic/automatic.dm [hunk 2]
```diff
@@ -164,11 +164,8 @@
 	underbarrel = new /obj/item/gun/ballistic/revolver/grenadelauncher/unrestricted(src)
 	update_icon()
 
-/obj/item/gun/ballistic/automatic/m90/afterattack(atom/target, mob/living/user, flag, params)
-	if(select == 2)
-		underbarrel.afterattack(target, user, flag, params)
-	else
-		return ..()
+/obj/item/gun/ballistic/automatic/m90/afterattack_secondary(atom/target, mob/living/user, flag, params)
+	underbarrel.afterattack(target, user, flag, params)
```
Comment (Jared-Fogle) [inline code/modules/projectiles/guns/ballistic/automatic.dm:168]: This shouldn't return null. You'll find that it'll runtime when you right click someone adjacent to you.

Return `SECONDARY_ATTACK_CONTINUE_CHAIN` instead.

**Comment by tgstation-server** [comment]
Your Fix/Feature pull request delta is currently below zero (-3). Maintainers may close future Feature/Tweak/Balance PRs. Fixing issues or helping to improve the codebase will raise this score.

**Pull request merged**